### PR TITLE
`CompiledTerm` to Plutarch `ClosedTerm`

### PR DESCRIPTION
**Pull request by SeungheonOh**
I think `toPTerm` needs a better name.

#51

### CODE REVIEW

File: src/Plutarch/Extra/Precompile.hs [hunk 1]
```diff
@@ -48,12 +49,14 @@ import Plutarch.Lift (
     PConstantDecl (pconstantFromRepr),
     PUnsafeLiftDecl (PLifted),
  )
-import Plutarch.Prelude (PLift, S, Term, Type, (:-->))
+import Plutarch.Prelude (PLift, S, Term, Type, (:-->), ClosedTerm)
+import Plutarch.Internal (Term (Term), TermResult(TermResult), RawTerm(RCompiled))
```
Comment (L-as) [inline src/Plutarch/Extra/Precompile.hs:53]: really big no IMO... what are you trying to do?

**Comment by L-as** [comment]
I see the point, but you want to use Plutarch.FFI here.

**Comment by SeungheonOh** [comment]
`unsafeForeignImport` ?

**Comment by SeungheonOh** [comment]
Okay, It's not using `Plutarch.FFI`. It's okay to use `unsafeForeignImport` because `CompiledTerm` will always have correct Plutarch type.

File: src/Plutarch/Extra/Precompile.hs [hunk 2]
```diff
@@ -196,13 +197,17 @@ infixl 7 ###
...
+ This can then be used to `plift`.
 -}
-toPTerm :: forall p. CompiledTerm p -> ClosedTerm p
-toPTerm prog = Term $ const $ pure $ TermResult (RCompiled $ UPLC._progTerm $ unScript . debugScript . toDScript $ prog) []
+toPTerm ::
+    forall (p :: S -> Type).
+    CompiledTerm p ->
+    ClosedTerm p
+toPTerm prog =
+    let (Script (UPLC.Program _ version term)) = debugScript . toDScript $ prog
```
Comment (cjay) [inline src/Plutarch/Extra/Precompile.hs:208]: Why does this use `debugScript`? Pretty sure it should use `finalEvalDScript`, like `pliftCompiled'`.

Comment (cjay) [reply src/Plutarch/Extra/Precompile.hs:208]: Hmm and unless we can get the official `plift` to output trace messages, maybe it should only use `script`. I'll ask if we could get that change in.

**Pull request closed without merging**